src/http.ts

```diff
@@ -298,16 +299,19 @@ const CREATE_GIT_BUNDLE_ERROR_MESSAGES: { [P in CreateGitBundleErrorCodes]: stri
   [ErrorCodes.notFound]: 'Unable to found requested repository or commit hash',
 };
 
-export async function createGitBundle(options: {
-  readonly baseURL: string;
-  readonly sessionToken: string;
-  readonly oAuthToken?: string;
-  readonly username?: string;
-  readonly gitUri: string;
-  readonly source: string;
-}): Promise<IResult<RemoteBundle, CreateGitBundleErrorCodes>> {
+export async function createGitBundle(
+  options: {
+    readonly baseURL: string;
+    readonly sessionToken: string;
+    readonly oAuthToken?: string;
+    readonly username?: string;
+    readonly gitUri: string;
+    readonly source: string;
+  },
+  requestOptions?: RequestOptions,
+): Promise<IResult<RemoteBundle, CreateGitBundleErrorCodes>> {
   const { baseURL, sessionToken, oAuthToken, username, gitUri, source } = options;
-  const headers = { 'Session-Token': sessionToken, source };
+  const headers = { ...requestOptions?.headers, 'Session-Token': sessionToken, source };
```

Comment:
I see `requestOptions` is an optional, what happens if they're missing?

Reply:
Nothing :-) Question mark is checking for it being available: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
Sounds good, thanks for sharing the docs 😀